src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1952,15 +1939,17 @@ MulticopterPositionControl::control_position(float dt)
 
 			} else {
 				/* Z component is negative, going down, simply limit thrust vector */
-				float k = thr_max / thrust_abs;
+				float k = thr_max / thrust_body_z;
 				thrust_sp *= k;
```

Comment:
The only error was the line I mentioned above but it had no functional effect.
Yes, `thrust_sp` gets used later on in the attitude setpoint calculation.
I did not rework the entire limiting logic yet but at least it works unaltered like before.